justfile

```
@@ -13,12 +13,41 @@ default_docker_tag := 'local'
 default_repo_name := 'ghcr.io/astriaorg'
 
 # Builds docker image for the crate. Defaults to 'local' tag.
-docker-build crate tag=default_docker_tag repo_name=default_repo_name:
-  docker buildx build --load --build-arg TARGETBINARY={{crate}} -f containerfiles/Dockerfile -t {{repo_name}}/$(echo {{crate}} | sed 's/^[^-]*-//'):{{tag}} .
-
-docker-build-and-load crate tag=default_docker_tag repo_name=default_repo_name:
-  @just docker-build {{crate}} {{tag}} {{repo_name}}
-  @just load-image $(echo {{crate}} | sed 's/^[^-]*-//') {{tag}} {{repo_name}}
+# NOTE: `_crate_short_name` is invoked as dependency of this command so that failure to pass a valid
+# crate will produce a maningful error message.
+docker-build crate tag=default_docker_tag repo_name=default_repo_name: (_crate_short_name crate "quiet")
+  #!/usr/bin/env sh
+  set -eu
+  short_name=`just _crate_short_name {{crate}}`
```

```suggestion
  short_name=$(just _crate_short_name {{crate}})
```
My comment re command substitution was due to my shell (fish), but setting `#!/usr/bin/env sh` fixes this (and backticks is probably more confusing nowadays than `$()`).

---

Disagree: just uses that for its own DSL (the portion you linked), while here we have an inline script and command substitution is considered better practice and is strictly preferred over backticks.